src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1352,31 +1609,46 @@ internal ProcessInputWriter(InternalCommand command)
         {
             Dbg.Assert(command != null, "Caller should validate the parameter");
             _command = command;
+
+            _pipeline = ScriptBlock.Create("Out-String -Stream").GetSteppablePipeline();
```

Comment:
Hmmm. Ok, so I would need to double check this. Here is my original thinking:

Before we had all data upfront, so we can use just `Out-String`. Now we get new data sporadically and should still output something, even when data is incomplete. `-Stream` seemed like a right switch.

After your comments, I revisited this thinking and it could be indeed incorrect. I think data would still be available, there would be no aggregation. For example, lets take a look at

```
$h = @{1 = 2}; $h; sleep 1; $h | Out-String | %{$_}
```

When I run it, I get the first `$h` immediately.

Also `-Stream` adds overhead, because there are more objects. So I think you are correct. Thank you for spotting the problem!